### ISSUE TITLE
Serverswitchevent on 1.7 Client crashes due to existing scoreboard teams

### ISSUE THREAD

**Issue by c0dingnoobi** [NEZNAMY/TAB#727]
### Server version

1.8.8

### TAB version

3.1.5

### Describe the bug

Players on 1.7 experience crashes sometimes when switching servers while their scoreboard is enabled. 
It is not happening often, sometimes it works without any problems. To reproduce u either have to relog then multitple times until it doesnt work or restart the servers. 

### Expected behavior

Server switching without crashes

### Steps to reproduce

1. Join on 1.8 server using 1.7
2. Have default config with scoreboards and globalplayerlist set
3. Switch servers to see crash

### Additional info

Client crash report: 
[crash-2022-10-09_00.33.12-client.txt](https://github.com/NEZNAMY/TAB/files/9747798/crash-2022-10-09_00.33.12-client.txt)

### Checklist

- [X] I am running latest version of the plugin
- [X] I have read the wiki to make sure it's not an issue with configuration
- [X] I have read the Compatibility wiki page and am not trying to run the plugin on an unsupported server version / platform

**Comment by BomBardyGamer** [comment]
Some important details have been omitted here:

1. You are using lunar client, which has some known issues with compatibility.
2. I assume that, as you are connecting to a 1.8 server with a 1.7 client, the server is running some form of protocol translation software, most likely ViaVersion with ViaRewind.
3. Details about the servers being switched between and some other multi-server setup details. Are you switching between two 1.8 servers running the same software? Is TAB running on both of the backend servers and/or the proxy (which I assume is BungeeCord or a derivative, as that's the only proxy we support)?

Also, you clearly haven't read the compatibility guide well enough, as the last point at the very bottom has information about custom clients, and mentions how Lunar Client has 3 known bugs.

I don't know if anyone else wants to try and debug this, but I certainly won't try and debug this, as you're using a proprietary custom client and a version that is not supported by any of the sources that I could retrieve mappings to look at the game's source with anymore, and it hasn't been for ages.

If you can reproduce the issue without Lunar Client, with just a regular vanilla 1.7 client, then it's probably a Lunar issue, and no attempt will be made to fix it on our end. If it still occurs with a vanilla 1.7 client, me or nez will look in to it, and see what we can do.

**Comment by NEZNAMY** [comment]
XCord has issues on 1.7 I believe, try another fork.

**Comment by c0dingnoobi** [comment]
As
Mentioned
It also has been tested using latest Bungeecord.

**Comment by NEZNAMY** [comment]
Latest bungeecord does not support 1.7.

**Comment by c0dingnoobi** [comment]
Had ViaVersion plus ViaRewind while doing that. I can test using FlameCord which has 1.7 support.

**Comment by NEZNAMY** [comment]
ViaVersion on bungeecord is an unfixable compatibility issue.

**Comment by c0dingnoobi** [comment]
Also seems to be replicateable on FlameCord (without any plugin on proxy).

**Comment by c0dingnoobi** [comment]
After further testing for like 2 weeks, i could reproduce that specific problem also when ever 1.7 player does not have a scoreboard defined (like f.e if he is in a world which doesnt have a scoreboard set) and stays afk for about 10 - 20 min. 
This also happens to 1.7 vanilla Minecraft. It has been tested using Xcord and Flamecord (ran both for like 3 days to be sure). 
Via is only on backends.
When the client crashes it produces same Crash log with same error as i mentioned above. 

My theory is that somehow even if the player doesnt have the scoreboard given by TAB, it still tries to either update or load a non existing scoreboard which causes the crashes.
I personally think that the server switch event itself isnt the cause. I have tested defining a scoreboard to the server where i crashed before when switiching to. Having a scoreboard on the server leads to no crash. Whenever i remove that test scoreboard the client is again able to crash. 

So to sum up 
Afking in world without scoreboard defined --> crash
Switching to a server which has no scoreboard defined --> crash
Switching to server which has scoreboard defined --> no crash

Hopefully these small details did help.

**Comment by BomBardyGamer** [comment]
The fact that this is to do with scoreboard teams indicates to me that the issue is with player tags (the stuff above/below your name tag), rather than the scoreboard, which is a good indication. These errors would indicate to me that there's a desync between TAB on the server and the clients (TAB and the server think a team doesn't exist to set the nametag data, so they create one, but the client thinks the team still exists).

**Comment by c0dingnoobi** [comment]
Yea im currently also experiencing these crashes without scoreboards which should confirm your statement.

**Comment by NEZNAMY** [comment]
Try https://github.com/NEZNAMY/TAB/actions/runs/3393807158

**Comment by c0dingnoobi** [comment]
Works, when having disable_entity_metadata_rewrite: true in waterfall.yml set to true. 
Otherwhise its still crashing for same crash reason.

**Comment by NEZNAMY** [comment]
Sounds about right. Pipeline injection is disabled for 1.7 clients, so Login packet sent to clear data cannot be detected, so plugin assumes one was sent on server switch.

**Comment by c0dingnoobi** [comment]
Ah, just a reminder. It doesn't crash on serverswitchevents anymore. But it still does when being in a world which doesn't have a scoreboard set. Crash reason is also same (team already exists).
It happens randomly so i guess it has something to do with any kind of refreshing.

**Comment by NEZNAMY** [comment]
What is the name of the team?

**Comment by c0dingnoobi** [comment]
01playernameA

**Comment by NEZNAMY** [comment]
That's from teams feature, not sidebar as you originally reported. I'll need steps to reproduce.

**Comment by c0dingnoobi** [comment]
Okay. I can try to do more tests to see what exactly causes it. But for now I can only say that it randomly happens with 1.7 when being in a world which doesnt have sb defined.

**Comment by NEZNAMY** [comment]
That's not related to scoreboard in any way.

**Comment by c0dingnoobi** [comment]
Okay will keep that in mind. Will just do some testing might report when i know more about it.

**Comment by c0dingnoobi** [comment]
Okay seems to happen whenever disable_entity_metadata_rewrite: true is true in waterfall.yml
It still random and i cant figure out what does cause the crash exactly. But it only happens when that is enabled.

**Comment by NEZNAMY** [comment]
Do you have teams disabled in some servers?

**Comment by c0dingnoobi** [comment]
Not rlly no. I once removed the default two and set
```disable-in-servers: []```
```disable-in-worlds: []```

**Comment by NEZNAMY** [comment]
Try https://github.com/NEZNAMY/TAB/actions/runs/3406131262

**Comment by c0dingnoobi** [comment]
kk

**Comment by c0dingnoobi** [comment]
Okay that might sound weird. 
The error message u built in never came. The clients still crashing. After further testing this seems to be irrelevant to waterfall.yml option since it happens regardless if true or false. 
The error message which usually should indicate which team was involved also doesnt seem to happen anymore. It just sends me the NullPointerException which i got once before (https://pastebin.com/riDqGkGY). 
I am still testing to see what triggers it.

**Comment by NEZNAMY** [comment]
Looks like I should add the check to objectives as well. However, finding the cause with steps to reproduce will help more.

**Comment by c0dingnoobi** [comment]
Working on it can't rlly give u any informations about reproducing yet

**Comment by c0dingnoobi** [comment]
Just happened again. All players were online using 1.7 clients. The one in a world without scoreboard got kicked. The one who were in a world with a scoreboard did not get kicked.

**Comment by NEZNAMY** [comment]
The team name contains palyer's name. What happened to that player at the time? Did the rank change so the sorting should change? Or prefix?

**Comment by c0dingnoobi** [comment]
Well all player's were "afk" they mostly did nothing or only walked. Can't think of anything that would either have changed the sorting order or the player's prefix.

**Comment by c0dingnoobi** [comment]
> Looks like I should add the check to objectives as well. However, finding the cause with steps to reproduce will help more.

I tried with localhosted network which has only TAB on bungee (flamecord) and bridge on backend (and via + viarewind). Same behaviour as before. When being in f.e nether while only world has scoreboard set, players are getting kicked after some time. 
If scoreboard is disabled nothing happens. If both dimensions have scoreboards set 1.7 doesnt crash while being in nether. 

There are no reproducing steps. 
I could basically set a scoreboard for every world to avoid 1.7 this bug, which would fix it for me. But not for other users of TAB who aren't familiar why this happens.

**Comment by NEZNAMY** [comment]
Since the last error you sent is different, I tried to fix that one. Try new build.

**Comment by c0dingnoobi** [comment]
![image](https://user-images.githubusercontent.com/97745777/202070037-2ea7e52a-4ba3-4d2c-b79e-ba818e198086.png)
I guess this is a success. Player was afk in a world which doesn't have a scoreboard

**Comment by nhanledev** [comment]
I tried to run the latest dev build and it keeps creating error.log file like this.

```
16.11.2022 - 16:38:02 - [TAB v3.1.5] Tried to modify non-existing team 256hibaramamoruA for player HibaraMamoru
16.11.2022 - 16:38:11 - [TAB v3.1.5] Tried to unregister non-existing team 0255hibaramamorA for player HibaraMamoru
```

**Comment by NEZNAMY** [comment]
Makes perfect sense, see the commit I pushed. I will investigate this harmless warn.

**Comment by c0dingnoobi** [comment]
I dont know if this would help, but it seems to happen every 20 minutes.
